x/mint/types/minter.go

```diff
@@ -1,80 +1,40 @@
 // provisions values.
-func NewMinter(inflation, annualProvisions sdk.Dec) Minter {
+func NewMinter(annualProvisions sdk.Dec) Minter {
 	return Minter{
-		Inflation:        inflation,
 		AnnualProvisions: annualProvisions,
 	}
 }
 
-// InitialMinter returns an initial Minter object with a given inflation value.
-func InitialMinter(inflation sdk.Dec) Minter {
-	return NewMinter(
-		inflation,
-		sdk.NewDec(0),
-	)
+// InitialMinter returns an initial Minter object
+func InitialMinter() Minter {
+	return NewMinter(sdk.NewDec(0))
 }
 
 // DefaultInitialMinter returns a default initial Minter object for a new chain
-// which uses an inflation rate of 13%.
 func DefaultInitialMinter() Minter {
-	return InitialMinter(
-		sdk.NewDecWithPrec(13, 2),
-	)
+	return InitialMinter()
 }
 
 // validate minter
 func ValidateMinter(minter Minter) error {
-	if minter.Inflation.IsNegative() {
-		return fmt.Errorf("mint parameter Inflation should be positive, is %s",
-			minter.Inflation.String())
-	}
 	return nil
 }
 
-// NextInflationRate returns the new inflation rate for the next hour.
-func (m Minter) NextInflationRate(params Params, bondedRatio sdk.Dec) sdk.Dec {
-	// The target annual inflation rate is recalculated for each previsions cycle. The
-	// inflation is also subject to a rate change (positive or negative) depending on
-	// the distance from the desired ratio (67%). The maximum rate change possible is
-	// defined to be 13% per year, however the annual inflation is capped as between
-	// 7% and 20%.
-
-	// (1 - bondedRatio/GoalBonded) * InflationRateChange
-	inflationRateChangePerYear := sdk.OneDec().
-		Sub(bondedRatio.Quo(params.GoalBonded)).
-		Mul(params.InflationRateChange)
-	inflationRateChange := inflationRateChangePerYear.Quo(sdk.NewDec(int64(params.BlocksPerYear)))
-
-	// adjust the new annual inflation for this next cycle
-	inflation := m.Inflation.Add(inflationRateChange) // note inflationRateChange may be negative
-	if inflation.GT(params.InflationMax) {
-		inflation = params.InflationMax
-	}
-	if inflation.LT(params.InflationMin) {
-		inflation = params.InflationMin
-	}
-
-	return inflation
-}
-
-// NextAnnualProvisions returns the annual provisions based on current total
-// supply and inflation rate.
-func (m Minter) NextAnnualProvisions(_ Params, totalSupply sdk.Int) sdk.Dec {
-	return m.Inflation.MulInt(totalSupply)
+// NextAnnualProvisions returns the annual provisions
+func (m Minter) NextAnnualProvisions(_ Params) sdk.Dec {
+	return m.AnnualProvisions.QuoInt(sdk.NewInt(2))
 }
 
-// BlockProvision returns the provisions for a block based on the annual
+// EpochProvision returns the provisions for a block based on the annual
 // provisions rate.
-func (m Minter) BlockProvision(params Params) sdk.Coin {
-	provisionAmt := m.AnnualProvisions.QuoInt(sdk.NewInt(int64(params.BlocksPerYear)))
+func (m Minter) EpochProvision(params Params) sdk.Coin {
+	provisionAmt := m.AnnualProvisions.QuoInt(sdk.NewInt(int64(params.EpochsPerYear)))
```

Comment:
I feel like this logic & querying would be improved if we instead made a "rewards per epoch" variable in state

Reply:
If we have "rewards per epoch" variable, we would have to merge EpochsPerYear, AnnualProvisions into one.
I am not sure why Cosmos SDK didn't do RewardsPerBlock - was following similar style of SDK.

Reply:
Ok that sounds good. If all tests are working atm, seems fine to leave it as is

Reply:
I think merging makes sense. Cosmos SDK did that because wanted to display inflation in terms of "annual inflation". But our framing is different, and "issuance per week" is okay, so merging these and getting rid of an extra variable makes sense imo.

Reply:
Ok, merging this would be good.